// Copyright (c) 2023 dingodb.com, Inc. All Rights Reserved
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "common.proto";

package dingodb.pb.store_internal;

option java_package = "io.dingodb.store_internal";
option cc_generic_services = true;

message SstFileInfo {
  int32 level = 1;
  string name = 2;
  string path = 3;
  bytes start_key = 4;
  bytes end_key = 5;
}

message RaftMeta {
  int64 region_id = 1;
  int64 term = 2;
  int64 applied_index = 3;
}

message Region {
  uint64 id = 1;
  dingodb.pb.common.RegionDefinition definition = 2;
  uint64 leader_id = 3;
  dingodb.pb.common.StoreRegionState state = 4;
  repeated dingodb.pb.common.StoreRegionState history_states = 5;
}